astropy/modeling/optimizers.py

```diff
@@ -240,3 +242,125 @@ def __call__(self, objfunc, initval, fargs, **kwargs):
                           "Maximum number of iterations reached.",
                           AstropyUserWarning)
         return fitparams, self.fit_info
+
+
+class Minimize(Optimization):
+    """
+    Interface for `scipy.optimize.minimize`.
+
+
+    Parameters
+    ----------
+    method : str or callable(custom)
+        Selects minimization method.
+
+    supported_constraints : list
+        The constraint types supported by the Optimization method. By default fixed and tied parameters are supported but
+        these can be set for custom methods. For defined methods this will be changed automatically.
+
+
+    """
+
+    jac_required = False
+
+    def __init__(self, method='slsqp', supported_constraints=None):
+        from scipy.optimize import minimize
+        super(Minimize, self).__init__(minimize)
+        method = method.lower()
+        if callable(method):
+            if supported_constraints is None:
+                warnings.warn("No supported_constraints set, by default the optimizer will only support fixed and tied parameters."
+                              "If you wish this optimizer to support more then you change this", AstropyUserWarning)
+
+                self.supported_constraints = ['fixed', 'tied']
+            else:
+                self.supported_constraints = supported_constraints
+        else:
+            if method in ['nelder-mead', 'powell', 'cg', 'bfgs', 'newton-cg', 'dogleg', 'trust-ncg']:
+                self.supported_constraints = ['fixed', 'tied']
+            elif method in ['l-bfgs-b', 'tnc']:
+                self.supported_constraints = ['fixed', 'tied', 'bounds']
+            elif method == 'cobyla':
+                self.supported_constraints = ['fixed', 'tied', 'ineqcons']
+            elif method in ['slsqp']:
+                self.supported_constraints = ['fixed', 'tied', 'eqcons', 'ineqcons', 'bounds']
+
+            if ['newton-cg', 'trust-ncg']:
```

Comment:
A few more trust-region algorithms were added in `scipy` recently; maybe you want to include them as well?

Reply:
Also I guess this line should be `if method in ...` ? Currently it will always be True ;).